hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/handlers/ContractCallLocalHandlerTest.java

```diff
@@ -0,0 +1,169 @@
+import static org.mockito.BDDMockito.given;
+
+import com.hedera.hapi.node.base.ContractID;
+import com.hedera.hapi.node.base.QueryHeader;
+import com.hedera.hapi.node.base.ResponseHeader;
+import com.hedera.hapi.node.contract.ContractCallLocalQuery;
+import com.hedera.hapi.node.state.token.Account;
+import com.hedera.hapi.node.transaction.Query;
+import com.hedera.node.app.service.contract.impl.exec.CallOutcome;
+import com.hedera.node.app.service.contract.impl.exec.ContextQueryProcessor;
+import com.hedera.node.app.service.contract.impl.exec.QueryComponent;
+import com.hedera.node.app.service.contract.impl.handlers.ContractCallLocalHandler;
+import com.hedera.node.app.service.token.ReadableAccountStore;
+import com.hedera.node.app.spi.workflows.PreCheckException;
+import com.hedera.node.app.spi.workflows.QueryContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class ContractCallLocalHandlerTest {
+    @Mock
+    private QueryContext context;
+
+    @Mock
+    private QueryComponent.Factory factory;
+
+    @Mock
+    private ContractCallLocalQuery contractCallLocalQuery;
+
+    @Mock
+    private QueryHeader header;
+
+    @Mock
+    private ResponseHeader responseHeader;
+
+    @Mock
+    private Query query;
+
+    @Mock
+    private ContractID contractID;
+
+    @Mock
+    private ReadableAccountStore store;
+
+    @Mock
+    private Account contract;
+
+    @Mock
+    private QueryComponent component;
+
+    @Mock
+    private ContextQueryProcessor processor;
+
+    private final ContractCallLocalHandler subject = new ContractCallLocalHandler(() -> factory);
+
+    @Test
+    void extractHeaderTest() {
+        // given:
+        given(query.contractCallLocalOrThrow()).willReturn(contractCallLocalQuery);
+        given(contractCallLocalQuery.header()).willReturn(header);
+
+        // when:
+        var header = subject.extractHeader(query);
+
+        // then:
+        assertThat(header).isNotNull();
+    }
+
+    @Test
+    void createEmptyResponseTest() {
+        // when:
+        var response = subject.createEmptyResponse(responseHeader);
+
+        // then:
+        assertThat(response).isNotNull();
+    }
+
+    @Test
+    void validatePositiveTest() throws PreCheckException {
+        // given
+        given(context.query()).willReturn(query);
+        given(query.contractCallLocalOrThrow()).willReturn(contractCallLocalQuery);
+        given(contractCallLocalQuery.contractID()).willReturn(contractID);
+        given(context.createStore(ReadableAccountStore.class)).willReturn(store);
+        given(store.getContractById(contractID)).willReturn(contract);
+
+        // when:
+        assertThatCode(() -> subject.validate(context)).doesNotThrowAnyException();
+    }
+
+    @Test
+    void validateFailsIfNoContractIdTest() throws PreCheckException {
```

Comment:
None of these `throws PreCheckException` are needed - the assertions eat the checked exceptions.